File: examples/javascript/openai-sdk.js
import OpenAI from 'openai';

const client = new OpenAI({
  baseURL: 'http://localhost:3000/v1',
  apiKey: 'anything',
});

async function main() {
  // 1. List models
  console.log('=== Models ===');
  const models = await client.models.list();
  console.log(models.data.slice(0, 3));

  // 2. Chat completion
  console.log('\n=== Chat Completion ===');
  const chat = await client.chat.completions.create({
    model: 'meta/llama-3.3-70b-instruct',
    messages: [{ role: 'user', content: 'Say hello in one word.' }],
    temperature: 0.7,
    max_tokens: 50,
  });
  console.log(`Response: ${chat.choices[0].message.content}`);
  console.log(`Usage: ${JSON.stringify(chat.usage)}`);

  // 3. Streaming
  console.log('\n=== Streaming ===');
  const stream = await client.chat.completions.create({
    model: 'meta/llama-3.3-70b-instruct',
    messages: [{ role: 'user', content: 'Count from 1 to 5.' }],
    stream: true,
  });
  for await (const chunk of stream) {
    process.stdout.write(chunk.choices[0]?.delta?.content || '');
  }
  console.log('\n');

  // 4. Embeddings
  console.log('=== Embeddings ===');
  const emb = await client.embeddings.create({
    model: 'nvidia/nv-embedqa-e5-v5',
    input: 'Hello world',
  });
  console.log(`Embedding dimension: ${emb.data[0].embedding.length}`);
  console.log(`Usage: ${JSON.stringify(emb.usage)}`);
}

main().catch(console.error);

File: examples/python/openai_sdk.py
from openai import OpenAI

client = OpenAI(
    base_url='http://localhost:3000/v1',
    api_key='anything',
)


def main():
    # 1. List models
    print('=== Models ===')
    models = client.models.list()
    for m in models.data[:3]:
        print(f"  {m.id}")

    # 2. Chat completion
    print('\n=== Chat Completion ===')
    chat = client.chat.completions.create(
        model='meta/llama-3.3-70b-instruct',
        messages=[{'role': 'user', 'content': 'Say hello in one word.'}],
        temperature=0.7,
        max_tokens=50,
    )
    print(f"Response: {chat.choices[0].message.content}")
    print(f"Usage: {chat.usage}")

    # 3. Streaming
    print('\n=== Streaming ===')
    stream = client.chat.completions.create(
        model='meta/llama-3.3-70b-instruct',
        messages=[{'role': 'user', 'content': 'Count from 1 to 5.'}],
        stream=True,
    )
    for chunk in stream:
        content = chunk.choices[0].delta.content or ''
        print(content, end='')
    print()

    # 4. Embeddings
    print('\n=== Embeddings ===')
    emb = client.embeddings.create(
        model='nvidia/nv-embedqa-e5-v5',
        input='Hello world',
    )
    print(f"Embedding dimension: {len(emb.data[0].embedding)}")
    print(f"Usage: {emb.usage}")


if __name__ == '__main__':
    main()
